source/documentation/deploying_apps/deploying_java.md

```diff
@@ -0,0 +1,122 @@
+                server.setHandler(new App());
+
+                server.start();
+                server.join();
+        }
+    }
+    ```
+4. Compile your class:
+
+    ```bash
+    javac -d . -cp ./jetty.jar App.java
+    ```
+
+5. Combine your class with Jetty to create an executable jar file:
+
+    ```bash
+    jar uvfe ./jetty.jar app.App app/App.class
+    ``` 
+
+6. Push your jar file.
+
+    ```bash
+    cf push your-app-name -b java_buildpack -p ./jetty.jar
+    ```
+
+You have now deployed your Java application. Your application should now be accessible over HTTPS from the URL provided in the output.
+
+If your application requires arguments in order to start, you can set the `JBP_CONFIG_JAVA_MAIN` environment variable. For example to add `server` and `config.yml` as arguments for your app, you could add the following to your application's `manifest.yml`:
+
+```yaml
+env:
+  JBP_CONFIG_JAVA_MAIN: '{ arguments: "server config.yml" }
+```
+
+### Deploying a WAR file
+
+If your application can be packaged as a `.war` file you can deploy with the following:
+
+```bash
+cf push your-app-name -b java_buildpack -p your-app.war
+```
+
+If you need to use a specific version of Tomcat you can set the environment variable `JBP_CONFIG_TOMCAT`. For example to use Tomcat 8 you could add the following to your application's `manifest.yml`:
+
+```yaml
+env:
+  JBP_CONFIG_TOMCAT: '{ tomcat: { version: 8.0.+ } }'
+```
+Note that you do **not** need to deploy Tomcat along with your application. The Java buildpack will run the servlet 2 and 3 web applications.
+
+For more configuration options see [Tomcat Container](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/container-tomcat.md) [external link].
+
+### Deploying other JVM-based applications
+
+The Java buildpack supports running any JVM-based applications with little or zero additional configuration. For detailed examples of deploying applications using other frameworks and JVM-based languages:
+
+* [Embedded web server](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-embedded-web-server.md) [external link]
+* [Grails](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-grails.md) [external link]
+* [Groovy](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-groovy.md) [external link]
+* [Java Main](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-java_main.md) [external link]
+* [Play Framework](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-play_framework.md) [external link]
+* [Servlet](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-servlet.md) [external link]
+* [Spring Boot CLI](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/example-spring_boot_cli.md) [external link]
+* [Cloud Foundry Java Buildpack documentation](https://github.com/cloudfoundry/java-buildpack) [external link]
```

Comment:
It's probably worth using the `#readme` anchor for this URL to jump straight down to the start of the docs.

Reply:
Changed https://github.com/cloudfoundry/java-buildpack to https://github.com/cloudfoundry/java-buildpack/blob/master/README.md